incubator/binding-openapi/src/main/java/io/aklivity/zilla/runtime/binding/openapi/internal/config/OpenapiOptionsConfigAdapter.java

```diff
@@ -0,0 +1,158 @@
+    public String type()
+    {
+        return OpenapiBinding.NAME;
+    }
+
+    @Override
+    public JsonObject adaptToJson(
+        OptionsConfig options)
+    {
+        OpenapiOptionsConfig openOptions = (OpenapiOptionsConfig) options;
+
+        JsonObjectBuilder object = Json.createObjectBuilder();
+
+        if (openOptions.tls != null)
+        {
+            final TlsOptionsConfig tls = ((OpenapiOptionsConfig) options).tls;
+            object.add(SPECS_NAME, tlsOptions.adaptToJson(tls));
+        }
+
+        HttpOptionsConfig http = openOptions.http;
+        if (http != null)
+        {
+            object.add(HTTP_NAME, httpOptions.adaptToJson(http));
+        }
+
+        if (openOptions.openapis != null)
+        {
+            JsonArrayBuilder keys = Json.createArrayBuilder();
+            openOptions.openapis.forEach(p -> keys.add(p.location));
+            object.add(SPECS_NAME, keys);
+        }
+
+        return object.build();
+    }
+
+    @Override
+    public OptionsConfig adaptFromJson(
+        JsonObject object)
+    {
+        OpenpaiOptionsConfigBuilder<OpenapiOptionsConfig> openapiOptions = OpenapiOptionsConfig.builder();
+
+        if (object.containsKey(TLS_NAME))
+        {
+            final JsonObject tls = object.getJsonObject(TLS_NAME);
+            final TlsOptionsConfig tlsOptions = (TlsOptionsConfig) this.tlsOptions.adaptFromJson(tls);
+            openapiOptions.tls(tlsOptions);
+        }
+
+        if (object.containsKey(HTTP_NAME))
+        {
+            JsonObject http = object.getJsonObject(HTTP_NAME);
+
+            final HttpOptionsConfig httpOptions = (HttpOptionsConfig) this.httpOptions.adaptFromJson(http);
+            openapiOptions.http(httpOptions);
+        }
+
+        if (object.containsKey(SPECS_NAME))
+        {
+            object.getJsonArray(SPECS_NAME).forEach(s -> openapiOptions.openapi(asOpenapi(s)));
+        }
+
+        return openapiOptions.build();
+    }
+
+    @Override
+    public void adaptContext(
+        ConfigAdapterContext context)
+    {
+        this.readURL = context::readURL;
+        this.tlsOptions = new OptionsConfigAdapter(Kind.BINDING, context);
+        this.tlsOptions.adaptType("tls");
+        this.httpOptions = new OptionsConfigAdapter(Kind.BINDING, context);
+        this.httpOptions.adaptType("http");
+    }
+
+    private OpenapiConfig asOpenapi(
+        JsonValue value)
+    {
+        final String location = ((JsonString) value).getString();
+        final String specText = readURL.apply(location);
+        OpenApi openapi = parseOpenApi(specText);
+
+        return new OpenapiConfig(location, openapi);
+    }
+
+    private OpenApi parseOpenApi(
+        String openapiText)
+    {
+        OpenApi openApi = null;
+        try (Jsonb jsonb = JsonbBuilder.create())
+        {
+            openApi = jsonb.fromJson(openapiText, OpenApi.class);
+        }
+        catch (Exception ex)
+        {
+            rethrowUnchecked(ex);
+        }
```

Comment:
Let's make sure we are using a validating schema to parse OpenAPI spec so we can communicate errors meaningfully.

Also, support OpenAPI spec 3.0, 3,1, perhaps also 2.0 since OpenAPI has been around for a long time?